articles/tech-lead-needed.md

```diff
@@ -0,0 +1,98 @@
+---
+authorName: Daniel Mölle
+authorGithubUsername: dmlfra
+title: Your team needs a Tech Lead, not a Lead Techie
+---
+# {{page.title}}
+
+In what follows, let us just assume that a tech lead is an experienced software engineer who is simultaneously (a) supposed to lead the development team and (b) responsible for the entire technical solution.
+
+I know that this is not the only valid interpretation. In the same way that different organizations have different habits with respect to setting up project teams, there is no universal definition for project roles such as *tech leads* or *software architects*. Moreover, I am well aware of the fact that many agile methodologies even discourage us from assigning such singular roles to individual team members (even though they often exist implicitly in that case). However, I can only write about my personal experience -- more than ten years in device development projects at Zühlke Frankfurt, all of which had a tech lead in the very sense outlined above.
+
+## The Lead Shock
+
+There are several points that tend to come up when you ask tech leads about their career experiences, but the most fundamental one seems to be that they were overwhelmed the first time they worked in that role (cf. *Talking with Tech Leads* by Patrick Kua). Why is that?
+
+In my humble opinion, there are two key factors.
+
+Firstly, stepping into the tech-lead role implies an explosion of responsibilities that a software engineer has never experienced beforehand (say, when being promoted from junior to senior developer, or whatever grades the organization defines, a few years earlier). And it gets worse: Many of these new responsibilities are non-technical and thus often very hard to grasp for someone with an engineering background. Being a tech lead does not only include obvious and natural shifts such as switching from *moderate-scale thinking* to *large-scale thinking* or switching from implementation work to concept work; it also includes switching from 90% hard facts to 70% soft subjects (YMMV) -- such as collaboration, communication, long-term risk management, expectation management, relationship management, et cetera.
```

Comment:
"say, when being promoted from junior to senior developer, or whatever grades the organization defines, a few years earlier" -> shorten to "say, when being promoted from junior to senior developer"

I do no get "(YMMV)"

Does mentioning soft subjects add value? I'd delete them.